draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -2009,6 +2009,33 @@ https://www.example.org
 </t>
 </section>
 
+<section title="TLS Connection Closure" anchor="tls.connection.closure">
+<t>
+   TLS provides a facility for secure connection closure. When a valid
+   closure alert is received, an implementation can be assured that no
+   further data will be received on that connection.  TLS
+   implementations &MUST; initiate an exchange of closure alerts before
+   closing a connection. A TLS implementation &MAY;, after sending a
+   closure alert, close the connection without waiting for the peer to
+   send its closure alert, generating an "incomplete close".  Note that
+   an implementation which does this &MAY; choose to reuse the session.
+   This &SHOULD; only be done when the application knows (typically
+   through detecting HTTP message boundaries) that it has received all
+   the message data that it cares about.
+</t>
+<t>
+   As specified in <xref target="RFC8446"/>, any implementation which receives a
+   connection close without first receiving a valid closure alert (a
+   "premature close") &MUST-NOT; reuse that session.  Note that a
```

Comment:
I don't know that I've noticed this requirement before.  Do implementations actually make this distinction?

Reply:
we can look into this when we trim this down later